src/modules/mavlink/mavlink_log_handler.cpp

```diff
@@ -315,47 +298,33 @@ MavlinkLogHandler::_log_send_data()
 {
 	mavlink_log_data_t response;
 	memset(&response, 0, sizeof(response));
-	uint32_t len = _pLogHandlerHelper->current_log_data_remaining;
+	uint32_t len = _current_log_data_remaining;
 
 	if (len > sizeof(response.data)) {
 		len = sizeof(response.data);
 	}
 
-	size_t read_size = _pLogHandlerHelper->get_log_data(len, response.data);
-	response.ofs     = _pLogHandlerHelper->current_log_data_offset;
-	response.id      = _pLogHandlerHelper->current_log_index;
+	size_t read_size = _get_log_data(len, response.data);
+	response.ofs     = _current_log_data_offset;
+	response.id      = _current_log_index;
 	response.count   = read_size;
 	mavlink_msg_log_data_send_struct(_mavlink->get_channel(), &response);
-	_pLogHandlerHelper->current_log_data_offset    += read_size;
-	_pLogHandlerHelper->current_log_data_remaining -= read_size;
+	_current_log_data_offset    += read_size;
+	_current_log_data_remaining -= read_size;
 
-	if (read_size < sizeof(response.data) || _pLogHandlerHelper->current_log_data_remaining == 0) {
-		_pLogHandlerHelper->current_status = LogListHelper::LOG_HANDLER_IDLE;
+	if (read_size < sizeof(response.data) || _current_log_data_remaining == 0) {
+		_current_status = LOG_HANDLER_STATE::IDLE;
 	}
 
 	return sizeof(response);
 }
 
 //-------------------------------------------------------------------
-LogListHelper::LogListHelper()
-	: next_entry(0)
-	, last_entry(0)
-	, log_count(0)
-	, current_status(LOG_HANDLER_IDLE)
-	, current_log_index(UINT16_MAX)
-	, current_log_size(0)
-	, current_log_data_offset(0)
-	, current_log_data_remaining(0)
-	, current_log_filep(nullptr)
-{
-	_init();
-}
-
-//-------------------------------------------------------------------
-LogListHelper::~LogListHelper()
+void MavlinkLogHandler::_close_and_unlink_files(void)
```

Comment:
clang tidy complains about this:
```
/tmp/jenkins/workspace/PX4_Firmware_PR-14452/src/modules/mavlink/mavlink_log_handler.cpp:323:49: error: redundant void argument list in function definition [modernize-redundant-void-arg,-warnings-as-errors]

[2020-05-13T17:02:25.307Z] void MavlinkLogHandler::_close_and_unlink_files(void)
```

```suggestion
void MavlinkLogHandler::_close_and_unlink_files()
```

Reply:
> I assume you tested this?

For the last changes, I made them without Hardware, so didn't check yet. wait for any merge. I will re-test it in few days